android/app/src/main/java/com/breez/client/plugins/breez/BreezBackup.java

```diff
@@ -0,0 +1,362 @@
+                                backupsMap.forEach((k, v) -> getNodeIdFolder(k));
+                            }
+                            else {
+                                m_result.success(backupsMap);
+                            }
+                        })
+                    .addOnFailureListener(m_activity, e -> {
+                        Log.e(TAG, "Error retrieving files", e);
+                        m_result.success(false);
+                    });
+            });
+    }
+
+    private void getNodeIdFolder(String nodeId) {
+        m_driveResourceClient
+            .getAppFolder()
+            .continueWithTask(task -> {
+                DriveFolder appFolder = task.getResult();
+                SortOrder descendingDateOrder = new SortOrder.Builder().addSortDescending(SortableField.MODIFIED_DATE).build();
+
+                Query query = new Query.Builder()
+                    .addFilter(Filters.eq(SearchableField.MIME_TYPE, DriveFolder.MIME_TYPE))
+                    .addFilter(Filters.eq(SearchableField.TITLE, nodeId))
+                    .setSortOrder(descendingDateOrder)
+                    .build();
+
+                Task<MetadataBuffer> queryTask = m_driveResourceClient.queryChildren(appFolder, query);
+                return  queryTask
+                    .addOnSuccessListener(m_activity,
+                        metadataBuffer -> {
+                            Log.w(TAG, metadataBuffer.toString());
+                            if (metadataBuffer.getCount() == 0) {
+                                m_result.success(false);
+                            }
+
+                            fetchBackupFiles(metadataBuffer.get(0).getDriveId().asDriveFolder());
+                        })
+                    .addOnFailureListener(m_activity, e -> {
+                        Log.e(TAG, "Error retrieving files", e);
+                        m_result.success(false);
+                    });
+            });
+    }
+
+    private void fetchBackupFiles(DriveFolder nodeIdFolder) {
+        List<String> backupPathsList = new ArrayList<>();
+        Query query = new Query.Builder()
+                .build();
+
+        Task<MetadataBuffer> queryTask = m_driveResourceClient.queryChildren(nodeIdFolder, query);
+
+        queryTask
+            .addOnSuccessListener(m_activity,
+                metadataBuffer -> {
+                    for (Metadata m : metadataBuffer) {
+                        m_driveResourceClient.openFile(m.getDriveId().asDriveFile(), DriveFile.MODE_READ_ONLY)
+                                .addOnSuccessListener(contents -> {
+                            DriveContents fileContents = contents;
+                            InputStream remoteFileInputStream = fileContents.getInputStream();
+
+                            File file = new File(m_activity.getCacheDir().getPath() + "/" + m.getTitle());
+                            FileOutputStream outputStream = null;
+                            try {
```

Comment:
instead of wrapping try/catch block many times, better to have one try/catch block that wraps the whole block of code. Also if the outputstream is not initialized correctly because of exception, there is no use to move on